proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/saml/HubResponseTranslator.java

```diff
@@ -43,9 +45,16 @@ public HubResponseTranslator(
     Response getTranslatedHubResponse(HubResponseContainer hubResponseContainer, CountryMetadataResponse countryMetadataResponse) {
         final List<EidasAttributeBuilder> eidasAttributeBuilders = new ArrayList<>();
 
-        final String pid = hubResponseContainer.getPid()
-                .map(p -> String.format(PID_PREFIX, countryMetadataResponse.getCountryCode(), p))
-                .orElse(null);
+        final String pid;
+        final SecureRandomIdentifierGenerationStrategy idGeneratorStrategy = new SecureRandomIdentifierGenerationStrategy();
```

Comment:
It may be easier to test this if it were set in a field on construction. In that way you could control its behaviour if you `@Mock`ed it. Worth debating with team whether its useful to do this for the sake of testing though.

Believe there's no chance of increased randomness if we instantiate every time, see https://stackoverflow.com/questions/295628/securerandom-init-once-or-every-time-it-is-needed